api/layer/multipart_upload.go

```diff
@@ -500,18 +505,29 @@ func (n *layer) getUploadParts(ctx context.Context, p *UploadInfoParams) (map[in
 			continue
 		}
 		info := objInfoFromMeta(p.Bkt, meta)
+		// skip objects which are completed by "complete-multipart-upload" because they have "s3-Upload-Id" attribute
+		if !isSystem(info) {
+			continue
+		}
 		numStr := info.Headers[UploadPartNumberAttributeName]
 		num, err := strconv.Atoi(numStr)
 		if err != nil {
 			return nil, errors.GetAPIError(errors.ErrInternalError)
 		}
 		res[num] = info
+		if err = n.systemCache.PutObject(systemObjectKey(p.Bkt, UploadPartName(p.UploadID, p.Key, num)), info); err != nil {
+			n.log.Warn("couldn't cache upload part", zap.Error(err))
+		}
+	}
+
+	if len(res) == 0 {
+		return nil, errors.GetAPIError(errors.ErrNoSuchUpload)
 	}
 
 	return res, nil
 }
 
-func createUploadPartName(uploadID, key string, partNumber int) string {
+func UploadPartName(uploadID, key string, partNumber int) string {
```

Comment:
I suppose we should add comment and/or change name to more clear (e.g.  `FormUploadPartName`).